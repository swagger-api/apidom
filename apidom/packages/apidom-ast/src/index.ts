// JSON AST related exports
export { default as JsonNode } from './json/nodes/JsonNode';
export { default as JsonDocument } from './json/nodes/JsonDocument';
export { default as JsonObject } from './json/nodes/JsonObject';
export { default as JsonProperty } from './json/nodes/JsonProperty';
export { default as JsonArray } from './json/nodes/JsonArray';
export { default as JsonValue } from './json/nodes/JsonValue';
export { default as JsonKey } from './json/nodes/JsonKey';
export { default as JsonString } from './json/nodes/JsonString';
export { default as JsonStringContent } from './json/nodes/JsonStringContent';
export { default as JsonEscapeSequence } from './json/nodes/JsonEscapeSequence';
export { default as JsonNumber } from './json/nodes/JsonNumber';
export { default as JsonTrue } from './json/nodes/JsonTrue';
export { default as JsonFalse } from './json/nodes/JsonFalse';
export { default as JsonNull } from './json/nodes/JsonNull';
export {
  isFalse as isJsonFalse,
  isProperty as isJsonProperty,
  isStringContent as isJsonStringContent,
  isEscapeSequence as isJsonEscapeSequence,
  isArray as isJsonArray,
  isKey as isJsonKey,
  isNull as isJsonNull,
  isNumber as isJsonNumber,
  isObject as isJsonObject,
  isString as isJsonString,
  isTrue as isJsonTrue,
} from './json/nodes/predicates';
// YAML AST related exports
export { default as YamlAlias } from './yaml/nodes/YamlAlias';
export { default as YamlCollection } from './yaml/nodes/YamlCollection';
export { default as YamlComment } from './yaml/nodes/YamlComment';
export { default as YamlDirective } from './yaml/nodes/YamlDirective';
export { default as YamlDocument } from './yaml/nodes/YamlDocument';
export { default as YamlKeyValuePair } from './yaml/nodes/YamlKeyValuePair';
export { default as YamlMapping } from './yaml/nodes/YamlMapping';
export { default as YamlNode } from './yaml/nodes/YamlNode';
export { default as YamlScalar } from './yaml/nodes/YamlScalar';
export { default as YamlSequence } from './yaml/nodes/YamlSequence';
export { default as YamlStream } from './yaml/nodes/YamlStream';
export { default as YamlTag } from './yaml/nodes/YamlTag';
export { default as YamlAnchor } from './yaml/nodes/YamlAnchor';
export {
  isAlias as isYamlAlias,
  isKeyValuePair as isYamlKeyValuePair,
  isDirective as isYamlDirective,
  isDocument as isYamlDocument,
  isMapping as isYamlMapping,
  isScalar as isYamlScalar,
  isSequence as isYamlSequence,
  isStream as isYamlStream,
  isTag as isYamlTag,
} from './yaml/nodes/predicates';
// generic AST related exports
export { default as Literal } from './Literal';
export { Point, default as Position } from './Position';
export { default as Error } from './Error';
export { default as ParseResult } from './ParseResult';
// AST traversal related exports
export { getVisitFn, BREAK, visit } from './visitor';
// YAML CST/AST transformers related exports
export {
  transform as transformTreeSitterYamlCST,
  keyMap as treeSitterYamlKeyMap,
} from './yaml/transformers/tree-sitter-yaml';
