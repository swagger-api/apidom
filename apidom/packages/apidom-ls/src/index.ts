// @ts-ignore
import ApiDOMParser from 'apidom-parser';
// @ts-ignore
import * as openapi3_1Adapter from 'apidom-parser-adapter-openapi-json-3-1';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export { default as getLanguageService } from './apidom-language-service';

export type {
  LanguageService,
  LanguageSettings,
  SymbolsContext,
  ValidationContext,
  CompletionContext,
  WorkspaceContextService,
  ColorsContext,
  LanguageServiceContext,
  SUPPORTED_LANGUAGES,
} from './apidom-language-types';

export const pocTODOToRemove: () => Promise<string> = async () => {
  const parser = ApiDOMParser();

  const value = `{
    "openapi": "3.0.0",
      "info": {
        "version": "0.1.9"
      }
    }`;
  parser.use(openapi3_1Adapter);
  // parser.use(asyncapi2_0Adapter);
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const parseResult = await parser.parse(value, { sourceMap: true });
  // console.log('Par', JSON.stringify(parseResult));
  return 'done';
};
