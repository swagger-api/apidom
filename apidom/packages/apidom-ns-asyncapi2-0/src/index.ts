export { default } from './namespace';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export {
  isAsycApi2_0Element,
  isAsycapiElement,
  isComponentsElement,
  isContactElement,
  isIdentifierElement,
  isInfoElement,
  isLicenseElement,
  isSchemaElement,
  isChannelsElement,
  isChannelItemElement,
  isServersElement,
  isServerElement,
} from './predicates';

export { default as AsyncapiElement } from './elements/Asyncapi';
export { default as AsyncApi2_0Element } from './elements/AsyncApi2-0';
export { default as ComponentsElement } from './elements/Components';
export { default as ContactElement } from './elements/Contact';
export { default as IdentifierElement } from './elements/Identifier';
export { default as InfoElement } from './elements/Info';
export { default as LicenseElement } from './elements/License';
export { default as SchemaElement } from './elements/Schema';
export { default as ChannelsElement } from './elements/Channels';
export { default as ChannelItemElement } from './elements/ChannelItem';
export { default as OperationElement } from './elements/Operation';
export { default as ParametersElement } from './elements/Parameters';
export { default as ChannelBindingsElement } from './elements/ChannelBindings';
export { default as ServersElement } from './elements/Servers';
export { default as ServerElement } from './elements/Server';
export { default as ServerVariableElement } from './elements/ServerVariable';
export { default as SecurityRequirementElement } from './elements/SecurityRequirement';
export { default as ServerBindingsElement } from './elements/ServerBindings';
