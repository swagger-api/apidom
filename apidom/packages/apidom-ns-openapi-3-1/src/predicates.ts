import { allPass, either, is } from 'ramda';
import { createPredicate } from 'apidom';

import ComponentsElement from './elements/Components';
import ContactElement from './elements/Contact';
import InfoElement from './elements/Info';
import LicenseElement from './elements/License';
import OpenapiElement from './elements/Openapi';
import OpenApi3_1Element from './elements/OpenApi3-1';
import SchemaElement from './elements/Schema';
import ServerElement from './elements/Server';
import ServerVariableElement from './elements/ServerVariable';
import PathsElement from './elements/Paths';
import PathItemElement from './elements/PathItem';
import OperationElement from './elements/Operation';
import ReferenceElement from './elements/Reference';

export const isOpenApiApi3_1Element = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq, hasClass }) => {
    const isElementTypeOpenApi3_1 = isElementType('openApi3-1');
    const primitiveEqObject = primitiveEq('object');
    const hasClassApi = hasClass('api');

    return either(
      is(OpenApi3_1Element),
      allPass([hasBasicElementProps, isElementTypeOpenApi3_1, primitiveEqObject, hasClassApi]),
    );
  },
);

export const isOpenapiElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeOpenapi = isElementType('openapi');
    const primitiveEqString = primitiveEq('string');

    return either(
      is(OpenapiElement),
      allPass([hasBasicElementProps, isElementTypeOpenapi, primitiveEqString]),
    );
  },
);

export const isInfoElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeInfo = isElementType('info');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(InfoElement),
      allPass([hasBasicElementProps, isElementTypeInfo, primitiveEqObject]),
    );
  },
);

export const isLicenseElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeLicense = isElementType('license');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(LicenseElement),
      allPass([hasBasicElementProps, isElementTypeLicense, primitiveEqObject]),
    );
  },
);

export const isContactElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeContact = isElementType('contact');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ContactElement),
      allPass([hasBasicElementProps, isElementTypeContact, primitiveEqObject]),
    );
  },
);

export const isComponentsElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeComponents = isElementType('components');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ComponentsElement),
      allPass([hasBasicElementProps, isElementTypeComponents, primitiveEqObject]),
    );
  },
);

export const isSchemaElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeSchema = isElementType('schema');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(SchemaElement),
      allPass([hasBasicElementProps, isElementTypeSchema, primitiveEqObject]),
    );
  },
);

export const isServerElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeServer = isElementType('server');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ServerElement),
      allPass([hasBasicElementProps, isElementTypeServer, primitiveEqObject]),
    );
  },
);

export const isServerVariableElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeServerVariable = isElementType('serverVariable');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ServerVariableElement),
      allPass([hasBasicElementProps, isElementTypeServerVariable, primitiveEqObject]),
    );
  },
);

export const isPathsElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypePaths = isElementType('paths');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(PathsElement),
      allPass([hasBasicElementProps, isElementTypePaths, primitiveEqObject]),
    );
  },
);

export const isPathItemElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypePathItem = isElementType('pathItem');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(PathItemElement),
      allPass([hasBasicElementProps, isElementTypePathItem, primitiveEqObject]),
    );
  },
);

export const isOperationElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeOperation = isElementType('operation');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(OperationElement),
      allPass([hasBasicElementProps, isElementTypeOperation, primitiveEqObject]),
    );
  },
);

export const isReferenceElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeReference = isElementType('reference');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ReferenceElement),
      allPass([hasBasicElementProps, isElementTypeReference, primitiveEqObject]),
    );
  },
);
