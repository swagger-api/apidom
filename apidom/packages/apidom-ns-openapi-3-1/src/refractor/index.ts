import { refract as baseRefract } from 'minim';
import { propOr } from 'ramda';
import { invokeArgs } from 'ramda-adjunct';
import { visit, Element, dereference, mergeAllVisitors, createNamespace } from 'apidom';

import specification from './specification';
import * as predicates from '../predicates';
import openApi3_1Namespace from '../namespace';
import { keyMap, getNodeType } from '../traversal/visitor';

const refract = <T extends Element>(
  value: any,
  { specPath = ['visitors', 'document', 'objects', 'OpenApi', '$visitor'], plugins = [] } = {},
): T => {
  const element = baseRefract(value);
  const resolvedSpec = dereference(specification);

  /**
   * This is where generic ApiDOM becomes semantic (namespace applied).
   * We don't allow consumers to hook into this translation.
   * Though we allow consumers to define their onw plugins on already transformed ApiDOM.
   */
  const rootVisitor = invokeArgs(specPath, [], resolvedSpec);
  // @ts-ignore
  visit(element, rootVisitor, { state: { specObj: resolvedSpec } });

  /**
   * Run plugins only when necessary.
   * Running plugins visitors means extra single traversal.
   * This can be optimized in future for performance.
   */
  if (plugins.length > 0) {
    const namespace = createNamespace(openApi3_1Namespace);
    const toolbox = { predicates: { ...predicates }, namespace };
    const pluginsSpecs = plugins.map((plugin: any) => plugin(toolbox));
    const pluginsVisitor = mergeAllVisitors(pluginsSpecs.map(propOr({}, 'visitor')), {
      // @ts-ignore
      nodeTypeGetter: getNodeType,
    });
    pluginsSpecs.forEach(invokeArgs(['pre'], []));
    const newElement: any = visit(rootVisitor.element, pluginsVisitor, {
      keyMap,
      // @ts-ignore
      nodeTypeGetter: getNodeType,
    });
    pluginsSpecs.forEach(invokeArgs(['post'], []));
    return newElement;
  }

  return rootVisitor.element;
};

export const createRefractor = (specPath: string[]) => (value: any, options = {}) =>
  refract(value, { specPath, ...options });

export default refract;
