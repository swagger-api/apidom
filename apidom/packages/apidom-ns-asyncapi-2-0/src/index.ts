import './refractor/registration';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export { default } from './namespace';

export {
  isAsyncApi2_0Element,
  isAsyncApiVersionElement,
  isChannelBindingsElement,
  isChannelItemElement,
  isChannelsElement,
  isComponentsElement,
  isContactElement,
  isIdentifierElement,
  isInfoElement,
  isLicenseElement,
  isOperationElement,
  isParameterElement,
  isParametersElement,
  isReferenceElement,
  isReferenceElementExternal,
  isSchemaElement,
  isSecurityRequirementElement,
  isServerElement,
  isServerBindingsElement,
  isServersElement,
  isServerVariableElement,
} from './predicates';

export {
  isAsyncApi2_0LikeElement,
  isParameterLikeElement,
  isReferenceLikeElement,
  isSchemaLikeElement,
  isSecurityRequirementLikeElement,
  isServerLikeElement,
  isAsyncApiExtension,
} from './refractor/predicates';

export { keyMap, getNodeType } from './traversal/visitor';

export { default as AsyncApi2_0Element } from './elements/AsyncApi2-0';
export { default as AsyncApiVersionElement } from './elements/AsyncApiVersion';
export { default as ChannelBindingsElement } from './elements/ChannelBindings';
export { default as ChannelItemElement } from './elements/ChannelItem';
export { default as ChannelsElement } from './elements/Channels';
export { default as ComponentsElement } from './elements/Components';
export { default as ContactElement } from './elements/Contact';
export { default as CorrelationIDElement } from './elements/CorrelationID';
export { default as DefaultContentTypeElement } from './elements/DefaultContentType';
export { default as IdentifierElement } from './elements/Identifier';
export { default as InfoElement } from './elements/Info';
export { default as LicenseElement } from './elements/License';
export { default as OperationElement } from './elements/Operation';
export { default as ParameterElement } from './elements/Parameter';
export { default as ParametersElement } from './elements/Parameters';
export { default as ReferenceElement } from './elements/Reference';
export { default as SchemaElement } from './elements/Schema';
export { default as SecurityRequirementElement } from './elements/SecurityRequirement';
export { default as ServerElement } from './elements/Server';
export { default as ServerBindingsElement } from './elements/ServerBindings';
export { default as ServersElement } from './elements/Servers';
export { default as ServerVariableElement } from './elements/ServerVariable';
