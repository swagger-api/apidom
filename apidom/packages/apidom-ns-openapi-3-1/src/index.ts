import './refractor/registration';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export { default } from './namespace';

export {
  isCallbackElement,
  isComponentsElement,
  isContactElement,
  isExternalDocumentationElement,
  isInfoElement,
  isJsonSchemaDialectElement,
  isLicenseElement,
  isOpenapiElement,
  isOpenApi3_1Element,
  isOperationElement,
  isParameterElement,
  isPathItemElement,
  isPathsElement,
  isReferenceElement,
  isReferenceElementExternal,
  isRequestBodyElement,
  isResponseElement,
  isResponsesElement,
  isSchemaElement,
  isSchemaElementExternal,
  isSecurityRequirementElement,
  isServerElement,
  isServerVariableElement,
  isMediaTypeElement,
} from './predicates';

export {
  isOpenApi3_1LikeElement,
  isParameterLikeElement,
  isReferenceLikeElement,
  isRequestBodyLikeElement,
  isResponseLikeElement,
  isServerLikeElement,
  isOpenApiExtension,
} from './refractor/predicates';

export { keyMap, getNodeType } from './traversal/visitor';

export { default as CallbackElement } from './elements/Callback';
export { default as ComponentsElement } from './elements/Components';
export { default as ContactElement } from './elements/Contact';
export { default as DiscriminatorElement } from './elements/Discriminator';
export { default as ExternalDocumentationElement } from './elements/ExternalDocumentation';
export { default as InfoElement } from './elements/Info';
export { default as JsonSchemaDialectVisitor } from './elements/JsonSchemaDialect';
export { default as LicenseElement } from './elements/License';
export { default as MediaTypeElement } from './elements/MediaType';
export { default as OpenapiElement } from './elements/Openapi';
export { default as OpenApi3_1Element } from './elements/OpenApi3-1';
export { default as OperationElement } from './elements/Operation';
export { default as ParameterElement } from './elements/Parameter';
export { default as PathItemElement } from './elements/PathItem';
export { default as PathsElement } from './elements/Paths';
export { default as ReferenceElement } from './elements/Reference';
export { default as RequestBodyElement } from './elements/RequestBody';
export { default as ResponseElement } from './elements/Response';
export { default as ResponsesElement } from './elements/Responses';
export { default as SchemaElement } from './elements/Schema';
export { default as SecurityRequirementElement } from './elements/SecurityRequirement';
export { default as ServerElement } from './elements/Server';
export { default as ServerVariableElement } from './elements/ServerVariable';
export { default as XmlElement } from './elements/Xml';
