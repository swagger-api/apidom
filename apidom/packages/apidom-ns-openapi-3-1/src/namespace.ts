import { NamespacePluginOptions } from 'minim';

import OpenApi3_1 from './elements/OpenApi3-1';
import Openapi from './elements/Openapi';
import Info from './elements/Info';
import License from './elements/License';
import Contact from './elements/Contact';
import Components from './elements/Components';
import Schema from './elements/Schema';
import Server from './elements/Server';
import ServerVariable from './elements/ServerVariable';
import Paths from './elements/Paths';
import PathItem from './elements/PathItem';
import Operation from './elements/Operation';
import Parameter from './elements/Parameter';
import Reference from './elements/Reference';
import ExternalDocumentation from './elements/ExternalDocumentation';
import RequestBody from './elements/RequestBody';
import Responses from './elements/Responses';
import Response from './elements/Response';
import Callback from './elements/Callback';
import SecurityRequirement from './elements/SecurityRequirement';
import { createRefractor } from './refractor';

// register refractors specific to element types
OpenApi3_1.refract = createRefractor(['visitors', 'document', 'objects', 'OpenApi', '$visitor']);
Openapi.refract = createRefractor([
  'visitors',
  'document',
  'objects',
  'OpenApi',
  'fixedFields',
  'Openapi',
]);
Info.refract = createRefractor(['visitors', 'document', 'objects', 'Info', '$visitor']);
License.refract = createRefractor(['visitors', 'document', 'objects', 'License', '$visitor']);
Contact.refract = createRefractor(['visitors', 'document', 'objects', 'Contact', '$visitor']);
Components.refract = createRefractor(['visitors', 'document', 'objects', 'Components', '$visitor']);
Schema.refract = createRefractor(['visitors', 'document', 'objects', 'Schema', '$visitor']);
Server.refract = createRefractor(['visitors', 'document', 'objects', 'Server', '$visitor']);
ServerVariable.refract = createRefractor([
  'visitors',
  'document',
  'objects',
  'ServerVariable',
  '$visitor',
]);
Paths.refract = createRefractor(['visitors', 'document', 'objects', 'Paths', '$visitor']);
PathItem.refract = createRefractor(['visitors', 'document', 'objects', 'PathItem', '$visitor']);
Operation.refract = createRefractor(['visitors', 'document', 'objects', 'Operation', '$visitor']);
Parameter.refract = createRefractor(['visitors', 'document', 'objects', 'Parameter', '$visitor']);
Reference.refract = createRefractor(['visitors', 'document', 'objects', 'Reference', '$visitor']);
ExternalDocumentation.refract = createRefractor([
  'visitors',
  'document',
  'objects',
  'ExternalDocumentation',
  '$visitor',
]);
RequestBody.refract = createRefractor([
  'visitors',
  'document',
  'objects',
  'RequestBody',
  '$visitor',
]);
Responses.refract = createRefractor(['visitors', 'document', 'objects', 'Responses', '$visitor']);
Response.refract = createRefractor(['visitors', 'document', 'objects', 'Response', '$visitor']);
Callback.refract = createRefractor(['visitors', 'document', 'objects', 'Callback', '$visitor']);
SecurityRequirement.refract = createRefractor([
  'visitors',
  'document',
  'objects',
  'SecurityRequirement',
  '$visitor',
]);

const openApi3_1 = {
  namespace: (options: NamespacePluginOptions) => {
    const { base } = options;

    base.register('openApi3_1', OpenApi3_1);
    base.register('openapi', Openapi);
    base.register('info', Info);
    base.register('license', License);
    base.register('contact', Contact);
    base.register('components', Components);
    base.register('schema', Schema);
    base.register('server', Server);
    base.register('serverVariable', ServerVariable);
    base.register('paths', Paths);
    base.register('pathItem', PathItem);
    base.register('operation', Operation);
    base.register('parameter', Parameter);
    base.register('reference', Reference);
    base.register('externalDocumentation', ExternalDocumentation);
    base.register('requestBody', RequestBody);
    base.register('responses', Responses);
    base.register('callback', Callback);
    base.register('securityRequirement', SecurityRequirement);
    base.register('response', Response);

    return base;
  },
};

export default openApi3_1;
