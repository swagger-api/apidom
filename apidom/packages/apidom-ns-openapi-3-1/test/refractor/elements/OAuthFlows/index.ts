import { expect } from 'chai';
import { sexprs } from 'apidom';

import { OAuthFlowsElement } from '../../../../src';

describe('refractor', function () {
  context('elements', function () {
    context('OAuthFlowsElement', function () {
      specify('should refract to semantic ApiDOM tree', function () {
        const oauthFlowsElement = OAuthFlowsElement.refract({
          implicit: {},
          password: {},
          clientCredentials: {},
          authorizationCode: {},
        });

        expect(sexprs(oauthFlowsElement)).toMatchSnapshot();
      });
    });
  });
});
