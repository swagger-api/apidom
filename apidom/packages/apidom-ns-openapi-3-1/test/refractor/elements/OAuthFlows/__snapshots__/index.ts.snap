// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refractor elements OAuthFlowsElement should refract to semantic ApiDOM tree 1`] = `
(OAuthFlowsElement
  (MemberElement
    (StringElement)
    (OAuthFlowElement))
  (MemberElement
    (StringElement)
    (OAuthFlowElement))
  (MemberElement
    (StringElement)
    (OAuthFlowElement))
  (MemberElement
    (StringElement)
    (OAuthFlowElement)))
`;
